script6.sh

```diff
@@ -0,0 +1,15 @@
+#!/bin/bash
+
+# Set the default value for NUMBER if not provided
+NUMBER=${1:-50}
+
+# Create a directory with a name based on the date
+DIRECTORY="zero_$(date +%Y%m%d%H%M%S)_files"
+mkdir "$DIRECTORY"
+
+# Generate the files
+for ((i=1; i<=$NUMBER; i++))
+do
+  FILENAME="File_${i}_.dat"
+  echo "0" | dd bs="$i" count=1 of="$DIRECTORY/$FILENAME"
```

Comment:
Notice that the demand was to save nullbytes (\x00) in the file and not ascii char 0 (\x30)